src/doc/topics/code-structure.md

```diff
@@ -0,0 +1,114 @@
+# Code and Project Structure
+
+## Code Structure
+
+We want to keep a standard code structure that we base on the original Reusable Chart API. The structure of a chart would look more or less like this:
+
+```
+  // D3 module requires
+  // helper requires
+  // Constants
+
+  // Data definition JSDoc comment
+  // Module definition JSDoc comment
+
+  return function module() {
+    // Private variables (more or less grouped)
+    // Extractor functions
+    // Format functions
+
+    function exports (_selection) {
+      ...
+      // Dimension setting
+      // Data cleaning with 'cleanData' function
+
+      // Main building blocks (optional)
+      buildScales();
+      buildAxis();
+      buildSVG();
+      drawGridLines();
+      drawElements();
+      drawAxis();
+
+      // Conditional building blocks
+    }
+
+    // Building block definitions
+    // API definitions ordered alphabetically
+  }
+
+```
+
+We like to do some light 'data cleaning' inside the `cleanData` method. That would usually include making sure that numbers are not strings. We also tend to use a reduce function, so we create a copy of the original data and cast only the usual quantities or values we need for creating the chart.
```

Comment:
"That includes type verification..."

"Cast only the required properties" makes a little kore sense I think